apimodels/perf.go

```diff
@@ -8,51 +8,82 @@ import (
-	BaseURL   string `json:"-"`
-	TaskID    string `json:"-"`
-	Execution int    `json:"_"`
+type GetPerfCountOptions struct {
+	SPSBaseURL   string `json:"-"`
+	CedarBaseURL string `json:"_"`
+	TaskID       string `json:"-"`
+	Execution    int    `json:"_"`
 }
 
-// CedarPerfCount holds one element, NumberOfResults, matching the json returned by
+// PerfCount holds one element, NumberOfResults, matching the json returned by
 // Cedar's perf count rest route.
-type CedarPerfCount struct {
+type PerfCount struct {
 	NumberOfResults int `json:"number_of_results"`
 }
 
-// CedarPerfResultsCount queries SPS for the number of perf results attached to a task.
-func CedarPerfResultsCount(ctx context.Context, opts GetCedarPerfCountOptions) (*CedarPerfCount, error) {
-	client := utility.GetHTTPClient()
-	defer utility.PutHTTPClient(client)
+// PerfResultsCount queries SPS for the number of perf results attached to a task.
+func PerfResultsCount(ctx context.Context, opts GetPerfCountOptions) (*PerfCount, error) {
+	// If the SPS base URL is set, use it to get the perf results count.
+	if opts.SPSBaseURL != "" {
+		client := utility.GetHTTPClient()
+		defer utility.PutHTTPClient(client)
+
+		requestURL := fmt.Sprintf("%s/raw_perf_results/tasks/%s/%d/count", opts.SPSBaseURL, url.PathEscape(opts.TaskID), opts.Execution)
+		response, err := client.Get(requestURL)
+		if err != nil {
+			return nil, errors.Wrap(err, "sending request to get perf results count")
+		}
+		if response.StatusCode != http.StatusOK {
+			body, err := io.ReadAll(response.Body)
+			if err == nil {
+				return nil, errors.Errorf("unexpected status code: %d: %s", response.StatusCode, body)
+			} else {
+				return nil, errors.Wrapf(err, "unexpected status code: %d", response.StatusCode)
+			}
+		}
 
-	requestURL := fmt.Sprintf("%s/raw_perf_results/tasks/%s/%d/count", opts.BaseURL, url.PathEscape(opts.TaskID), opts.Execution)
-	response, err := client.Get(requestURL)
-	if err != nil {
-		return nil, errors.Wrap(err, "sending request to get perf results count")
-	}
-	if response.StatusCode != http.StatusOK {
 		body, err := io.ReadAll(response.Body)
-		if err == nil {
-			return nil, errors.Errorf("unexpected status code: %d: %s", response.StatusCode, body)
-		} else {
-			return nil, errors.Wrapf(err, "unexpected status code: %d", response.StatusCode)
+		if err != nil {
+			return nil, errors.Wrap(err, "reading response body when getting perf results count")
+		}
+		testCount := &PerfCount{}
+		if err = json.Unmarshal(body, testCount); err != nil {
+			return nil, errors.Wrap(err, "unmarshaling perf results")
 		}
-	}
 
-	body, err := io.ReadAll(response.Body)
-	if err != nil {
-		return nil, errors.Wrap(err, "reading response body when getting perf results count")
-	}
-	testCount := &CedarPerfCount{}
-	if err = json.Unmarshal(body, testCount); err != nil {
-		return nil, errors.Wrap(err, "unmarshaling test results from Cedar")
+		return testCount, nil
+	} else {
```

Comment:
nit: since the above block always returns something this else block can be unwrapped